File: solutions/python/bob/1/bob.py
def response(hey_bob: str) -> str:
    hey_bob = hey_bob.strip()
    is_question = hey_bob.endswith("?")
    is_yelling = hey_bob.isupper() and any(c.isalpha() for c in hey_bob)

    if len(hey_bob) == 0:
        return "Fine. Be that way!"
    if is_yelling and not is_question:
        return "Whoa, chill out!"
    elif is_question and not is_yelling:
        return "Sure."
    elif is_yelling and is_question:
        return "Calm down, I know what I'm doing!"
    else:
        return "Whatever."

    return ""

File: solutions/cpp/bob/1/bob.cpp
#include "bob.h"
#include <algorithm>
#include <cctype>
#include <pstl/glue_algorithm_defs.h>

namespace bob {
const std::string hey(std::string const &str) {
  auto strx = str;
  strx.erase(std::remove_if(strx.begin(), strx.end(),
                            [](const char c) { return std::isspace(c); }),
             strx.end());

  if (strx.empty())
    return "Fine. Be that way!";

  bool question = strx.back() == '?';
  bool alpha_exist{};
  bool lower_exist{};

  for (auto c : strx) {
    if (std::isalpha(c))
      alpha_exist = true;
    if (std::islower(c))
      lower_exist = true;
  }

  if (question && !lower_exist && alpha_exist) {
    return "Calm down, I know what I'm doing!";
  }

  if (question)
    return "Sure.";

  if (!lower_exist && alpha_exist)
    return "Whoa, chill out!";

  return "Whatever.";
}

} // namespace bob
